src/test/java/database/controller/DatabaseServiceRestControllerTest.java

```diff
@@ -0,0 +1,224 @@
+        when(databaseService.addNewRecordToTable(entity)).thenReturn(newEntity);
+
+        controller.handleAddNewRecord(ctx);
+        verify(ctx).json(newEntity);
+    }
+
+    @Test
+    void POST_to_add_new_record_returns_400_when_database_not_exists() {
+        String entityClass = "Student";
+        Student entity = new Student();
+
+        when(ctx.pathParam("entityClass")).thenReturn(entityClass);
+        when(ctx.bodyAsClass(Student.class)).thenReturn(entity);
+        when(databaseService.addNewRecordToTable(entity)).thenThrow(BadRequestResponse.class);
+
+        assertThrows(BadRequestResponse.class, () -> controller.handleAddNewRecord(ctx));
+    }
+
+    @Test
+    void GET_to_get_all_records_returns_json_with_entities() {
+        String entityClass = "Student";
+        Iterable<BaseEntity> entities = List.of(new Student());
+
+        when(ctx.pathParam("entityClass")).thenReturn(entityClass);
+        when(databaseService.getAllRecordsFromTable(Student.class)).thenReturn(entities);
+
+        controller.handleGetAllRecords(ctx);
+        verify(ctx).json(entities);
+    }
+
+    @Test
+    void GET_to_get_all_records_returns_500_when_database_not_exist() {
+        String entityClass = "Student";
+
+        when(ctx.pathParam("entityClass")).thenReturn(entityClass);
+        when(databaseService.getAllRecordsFromTable(Student.class)).thenThrow(InternalServerErrorResponse.class);
+
+        assertThrows(InternalServerErrorResponse.class, () -> controller.handleGetAllRecords(ctx));
+    }
+
+    @Test
+    void PUT_to_update_record_returns_json_with_entity() {
+        String entityClass = "Student";
+        Student entity = new Student();
+        entity.setId(1);
+        Student updatedEntity = new Student();
+
+        when(ctx.pathParam("entityClass")).thenReturn(entityClass);
+        when(ctx.bodyAsClass(Student.class)).thenReturn(entity);
+        when(databaseService.updateRecordInTable(entity, 1)).thenReturn(updatedEntity);
+
+        controller.handleUpdateRecord(ctx);
+        verify(ctx).json(updatedEntity);
+    }
+
+    @Test
+    void PUT_to_update_record_returns_400_when_id_not_exist() {
+        String entityClass = "Student";
+        Student entity = new Student();
+        entity.setId(1);
+
+        when(ctx.pathParam("entityClass")).thenReturn(entityClass);
+        when(ctx.bodyAsClass(Student.class)).thenReturn(entity);
+        when(databaseService.updateRecordInTable(entity, 1)).thenThrow(BadRequestResponse.class);
+
+        assertThrows(BadRequestResponse.class, () -> controller.handleUpdateRecord(ctx));
+    }
+
+    @Test
+    void DELETE_to_remove_record_returns_true() {
+        String entityClass = "Student";
+        int id = 1;
+        Student entity = new Student();
+        entity.setId(id);
+
+        when(ctx.pathParam("entityClass")).thenReturn(entityClass);
+        when(ctx.pathParam("id")).thenReturn(String.valueOf(id));
```

Comment:
Please use constant from `DatabaseServiceRestController`. Don't forget to reorder variables in `DatabaseServiceRestController` since constant will not be private variable any more.